webapp/channels/src/components/admin_console/system_user_detail/system_user_detail.test.tsx

```diff
@@ -133,6 +135,203 @@ describe('SystemUserDetail', () => {
 
         expect(container).toMatchSnapshot();
     });
+
+    describe('change detection', () => {
+        test('should detect email changes and enable save', async () => {
+            const setNavigationBlockedMock = jest.fn();
+            const props = {
+                ...defaultProps,
+                setNavigationBlocked: setNavigationBlockedMock,
+                intl: {
+                    formatMessage: jest.fn().mockImplementation(({defaultMessage}) => defaultMessage),
+                } as MockIntl,
+            };
+            renderWithContext(<SystemUserDetail {...props}/>);
+
+            await waitFor(() => {
+                expect(screen.queryByTestId('loadingSpinner')).not.toBeInTheDocument();
+            });
+
+            const emailInputs = screen.getAllByDisplayValue(user.email);
+            const emailInput = emailInputs.find((input) => !(input as HTMLInputElement).disabled);
+
+            if (emailInput) {
+                fireEvent.change(emailInput, {target: {value: 'newemail@example.com'}});
```

Comment:
This `fireEvent` here isn't going to work after the React 18 upgrade because it's not wrapped in an `act` or a `userEvent` helper. Without `act`, React won't process the result of the change event or update the DOM in response to it.

Instead, I'd suggest changing this to use `userEvent.type` which will trigger the change event and wrap it in an `act` automatically